class/practice4.1/src/builder.cpp

```diff
@@ -0,0 +1,51 @@
+#include "builder.hpp"
+#include <map>
+#include <fstream>
+#include <string>
+#include <cstdlib>
+#include <format>
+Pizza::Pizza() :price(0.0) {}
+
+std::string Pizza::showPizza() const {
+    return std::format("First: {} Second: {} Third: {} Fourth: {} Price: ${}",
+                       first, second, third, fourth, price);
+}
+BestPizzaBuilder::BestPizzaBuilder() {
+    this->pizza = new Pizza();
+}
+
+
+void BestPizzaBuilder::setFirst(const std::string& firstPicked, const double cost) {
+    pizza->first = firstPicked;
+    pizza->price += cost;
+}
+
+void BestPizzaBuilder::setSecond(const std::string& secondPicked, const double cost) {
+    pizza->second = secondPicked;
+    pizza->price += cost;
+}
+
+void BestPizzaBuilder::setThird(const std::string& thirdPicked, const double cost) {
+    pizza->third = thirdPicked;
+    pizza->price += cost;
+}
+
+void BestPizzaBuilder::setFourth(const std::string& fourthPicked, const double cost) {
+    pizza->fourth = fourthPicked;
+    pizza->price += cost;
+}
+
+Pizza* BestPizzaBuilder::getPizza() {

```

Comment:
in this method we should recreate pizza object to build the next one. Therefore it is better to have some smart pointer for example `std::unique_ptr<Pizza>` and std::move semantics
`[[nodiscard]] std::unique_ptr<Pizza>&& getPizza`
another option is to just have `[[nodiscard]] Pizza&& getPizza` and plain non-pointer member.

Also your implementation has potential memory leaks as we have no destructor and delete operator so in case if no one asks for the pizza then corresponding memory will leak